frontend/src/metabase/lib/api.js

```diff
@@ -85,6 +85,10 @@ export class Api extends EventEmitter {
           ? { Accept: "application/json", "Content-Type": "application/json" }
           : {};
 
+        if (options.formData && options.fetch) {
+          delete headers["Content-Type"];
+        }
```

Comment:
Currently, our API wrapper only supports JSON data. Even for whitelabel icon uploads we serialize image data into a JSON. I think the more common/idiomatic pattern for file uploads is to use multipart/form-data, and this adds support for that via the `formData` flag on the endpoint definition.